### PR TITLE
Ct 859 add acts as tree with team import

### PR DESCRIPTION
**Pull request by stephenrichards**
This PR creates the rake task data:migrate:import:teams to import live
hierarchical team data from a CSV file.

the CSV file is located in the Confluence pages - download it to a convenient location
and change line 5 of lib/rake_task_helpers/team_importer.rb to read from that location.

### CODE REVIEW

File: app/models/category.rb [hunk 1]
```diff
@@ -19,4 +19,11 @@ class Category < ApplicationRecord
 
   has_many :cases
 
+  def self.foi
+    where(abbreviation: 'FOI').first
```
Comment (misaka) [inline app/models/category.rb]: I think `find_by!(abbreviation: 'FOI')` would be better. More succinct and it'll raise if it's not there.

File: app/models/category.rb [hunk 2]
```diff
@@ -19,4 +19,11 @@ class Category < ApplicationRecord
...
+  end
+
+  def self.gq
+    where(abbreviation: 'GQ').first
```
Comment (misaka) [inline app/models/category.rb]: Same comment as for `Category.foi`.

File: lib/rake_task_helpers/team_importer.rb [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  end
+
+  def process_business_unit(row)
+    puts "        Business Unit #{row[2]}"
+    @bu.save! unless @bu.nil?
+    _bg, _dir, bu, lead, area = row
+    @bu = BusinessUnit.new(parent_id: @dir.id, name: bu, email: default_email(bu))
+    @bu.properties << TeamProperty.new(key: 'lead', value: lead)
+    @bu.properties << TeamProperty.new(key: 'area', value: area) if area.present?
+    @bu.properties << TeamProperty.new(key: 'can_allocae', value: 'FOI')
```
Comment (misaka) [inline lib/rake_task_helpers/team_importer.rb]: Key should be `can_allocate` presumably.

File: lib/rake_task_helpers/team_importer.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      bg, dir, bu, _lead, _area  = row
+      if bg.present?
+        :business_group
+      elsif dir.present?
+        :directorate
+      elsif bu.present?
+        :business_unit
+      else
+        :area
+      end
```
Comment (misaka) [inline lib/rake_task_helpers/team_importer.rb]: Are we going to use this more than once? i.e. will we have to update the list a couple times before we provide an admin interface? If so, it seems we'd be safer with:

```
elsif area.present?
  :area
else
  raise "Unrecognised row: #{row}"
end
```

Comment (stephenrichards) [inline lib/rake_task_helpers/team_importer.rb]: No, we're not.  And we have complete control over the data.

File: lib/rake_task_helpers/team_importer.rb [hunk 5]
```diff
@@ -0,0 +1,105 @@
+require 'csv'
+
+class TeamImporter
+
+  FILENAME = File.join(ENV['HOME'], 'Dropbox', 'team_data.csv')
```
Comment (aliuk2012) [inline lib/rake_task_helpers/team_importer.rb]: Why is there a mention of Dropbox? Couldn't we just use a directory within the projects root?

**Pull request merged**